packages/boxel-ui/test-app/tests/integration/components/resizable-panel-group-horizontal-test.gts

```diff
@@ -0,0 +1,310 @@
+  return new Promise(function (resolve) {
+    setTimeout(resolve, ms);
+  });
+}
+
+class PanelProperties {
+  @tracked lengthPx?: number;
+  @tracked isHidden?: boolean;
+  @tracked defaultLengthFraction?: number;
+  @tracked minLengthPx?: number;
+
+  innerContainerStyle?: string;
+  outerContainerStyle?: string;
+  showResizeHandle?: boolean;
+  constructor(
+    panelArgs: {
+      lengthPx?: number;
+      isHidden?: boolean;
+      defaultLengthFraction?: number;
+      minLengthPx?: number;
+      innerContainerStyle?: string;
+      outerContainerStyle?: string;
+      showResizeHandle?: boolean;
+    } = {},
+    testArgs: {
+      innerContainerStyle?: string;
+      outerContainerStyle?: string;
+      showResizeHandle?: boolean;
+    } = {},
+  ) {
+    let {
+      lengthPx,
+      isHidden = false,
+      defaultLengthFraction,
+      minLengthPx,
+    } = panelArgs;
+    let { innerContainerStyle, outerContainerStyle, showResizeHandle } =
+      testArgs;
+    this.lengthPx = lengthPx;
+    this.isHidden = isHidden;
+    this.defaultLengthFraction = defaultLengthFraction;
+    this.minLengthPx = minLengthPx;
+
+    this.showResizeHandle = showResizeHandle;
+
+    this.innerContainerStyle = innerContainerStyle;
+    this.outerContainerStyle = outerContainerStyle;
+  }
+}
+
+class RenderController {
+  @tracked containerStyle = '';
+  @tracked panels: PanelProperties[] = [];
+}
+
+interface MyTestContext extends RenderingTestContext {
+  renderController: RenderController;
+}
+
+module('Integration | ResizablePanelGroup | horizontal', function (hooks) {
+  setupRenderingTest(hooks);
+  hooks.beforeEach(function (this: MyTestContext) {
+    this.renderController = new RenderController();
+    this.renderController.panels = [
+      new PanelProperties(
+        { defaultLengthFraction: 0.6 },
+        {
+          showResizeHandle: true,
+          outerContainerStyle:
+            'border: 1px solid red; height: 100%; overflow-y:auto',
+        },
+      ),
+      new PanelProperties(
+        { defaultLengthFraction: 0.4, minLengthPx: 50 },
+        {
+          outerContainerStyle: 'border: 1px solid red; height: 100%',
+        },
+      ),
+    ];
+  });
+
+  async function renderHorizontalResizablePanelGroup(
+    renderController: RenderController,
+  ) {
+    await render(<template>
+      {{! template-lint-disable no-inline-styles }}
+      <div id='test-container' style={{renderController.containerStyle}}>
+        <ResizablePanelGroup
+          @orientation='horizontal'
+          @reverseCollapse={{true}}
+          as |ResizablePanel ResizeHandle|
+        >
+          {{#each renderController.panels as |panel index|}}
+
+            <ResizablePanel
+              @defaultLengthFraction={{panel.defaultLengthFraction}}
+              @lengthPx={{panel.lengthPx}}
+              @minLengthPx={{panel.minLengthPx}}
+              @isHidden={{panel.isHidden}}
+            >
+              {{#if (eq index 1)}}
```

Comment:
Do we need this if condition

Reply:
no, fixed in 6fd71c7, thanks.

Implementation note: since (almost?) all of these test comments are about something that I copypasted and converted to horizontal orientation, I did finally extract the vertical/horizontal parameters and expected properties into a generic test. It’s more abstract to read but now there aren’t two parallel tests to maintain.